src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java

```diff
@@ -129,8 +149,56 @@ private String getAlertElement(VulnType currentVT, String element) {
 			case HSTS_MAX_AGE_DISABLED:
 				elementValue=Constant.messages.getString(MESSAGE_PREFIX + "max.age." + element);
 				break;
+			case HSTS_MULTIPLE_HEADERS:
+				elementValue=Constant.messages.getString(MESSAGE_PREFIX + "compliance.mutiple.header." + element);
+				break;
+			case HSTS_ON_PLAIN_RESP:
+				elementValue=Constant.messages.getString(MESSAGE_PREFIX + "plain.resp." + element);
+				break;
+			case HSTS_MAX_AGE_MISSING:
+				elementValue=Constant.messages.getString(MESSAGE_PREFIX + "compliance.max.age.missing." + element);
+				break;
+			case HSTS_META:
+					elementValue=Constant.messages.getString(MESSAGE_PREFIX + "compliance.meta." + element);
+					break;
 		}
 		return elementValue;
 	}
+	
+	private int getRisk(VulnType currentVT) {
+		switch (currentVT) {
+			case HSTS_MISSING:
+			case HSTS_MAX_AGE_DISABLED:
+			case HSTS_MULTIPLE_HEADERS:
+			case HSTS_MAX_AGE_MISSING:
+			case HSTS_META:
+				return(Alert.RISK_LOW);
+			case HSTS_ON_PLAIN_RESP:
+			default:
+				return(Alert.RISK_INFO);
+		}
+	}
+	
+	/**
+	 * Checks the source of the response for HSTS being set via a META tag which is explicitly
+	 * not supported per the spec (rfc6797). 
+	 * 
+	 * @param source the source of the response to be analyzed.
+	 * @return returns a string if HSTS was set via META (for use as alert evidence) otherwise return {@code null}.
+	 * @see https://tools.ietf.org/html/rfc6797#section-8.5
+	 */
+	private String getMetaHSTSEvidence(Source source) {
+		List<Element> metaElements = source.getAllElements(HTMLElementName.META);
+		String httpEquiv;
 
+		if (metaElements != null) {
+			for (Element metaElement : metaElements) {
+				httpEquiv = metaElement.getAttributeValue("http-equiv");
+				if (httpEquiv.equalsIgnoreCase("strict-transport-security")) {

```

Comment:
There's a NullPointerException when the meta tag does not have the attribute "http-equiv".
It could do the check as `"strict-transport-security".equalsIgnoreCase(httpEquiv)` instead, would take care of the null handling.